src/Medidata.ZipkinTracer.Core/ZipkinClient.cs

```diff
@@ -46,7 +46,7 @@ public ZipkinClient(ILog logger, IZipkinConfig zipkinConfig, IOwinContext contex
                     spanCollector,
                     new ServiceEndpoint(),
                     zipkinConfig.NotToBeDisplayedDomainList,
-                    zipkinConfig.Domain);
+                    zipkinConfig.Domain(context.Request));
```

Comment:
The `Func` makes it more flexible and allows consumers to build up this configuration parameter from elements of the request that are not just the URI. For example, at JE, we would prefer to use our logical service name here, rather than the incoming hostname, because our traffic can talk to things on more than one hostname.